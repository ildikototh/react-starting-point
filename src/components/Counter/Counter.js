import React, { useState, useEffect } from 'react';
// This is a stateless component example using hook
const Counter = () => {
  const [count, setCount] = useState(0);
  useEffect(() => {
    // Update the document title using the browser API
    document.title = `You clicked ${count} times`;
  });

  return (
    <div>
      <p>
        You clicked
        {' '}
        {count}
        {' '}
        times
      </p>
      <button type="button" data-testid="incrementButton" onClick={() => setCount(count + 1)}>
        Click me
      </button>
    </div>
  );
};

export default Counter;
