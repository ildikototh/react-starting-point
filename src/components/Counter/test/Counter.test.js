import React from 'react';
import { mount } from 'enzyme';
import Counter from '../Counter';

describe('Counter component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(<Counter />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a default counter value on the p tag', () => {
    expect(wrapper.find('p').text()).toEqual('You clicked 0 times');
  });

  it('should increase the counter clicking on the button', () => {
    wrapper.find('button').simulate('click');
    expect(wrapper.find('p').text()).toEqual('You clicked 1 times');
  });
});
