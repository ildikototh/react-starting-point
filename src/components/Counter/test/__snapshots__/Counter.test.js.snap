// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component should match the snapshot 1`] = `
<Counter>
  <div>
    <p>
      You clicked
       
      0
       
      times
    </p>
    <button
      data-testid="incrementButton"
      onClick={[Function]}
      type="button"
    >
      Click me
    </button>
  </div>
</Counter>
`;
